src/aws/cis-1.2.0/rules/aws-cis-1.2.0-3.6.ts

```diff
@@ -118,61 +121,47 @@ export default {
   }`,
   resource: 'queryawsAccount[*]',
   severity: 'medium',
-  conditions: {
-    path: '@.cloudtrail',
-    array_any: {
-      and: [
-        {
-          path: '[*].isMultiRegionTrail',
-          equal: 'Yes',
-        },
-        {
-          path: '[*].status.isLogging',
-          equal: true,
-        },
-        {
-          path: '[*].eventSelectors',
-          array_any: {
-            and: [
-              { path: '[*].readWriteType', equal: 'All' },
-              {
-                path: '[*].includeManagementEvents',
-                equal: true,
-              },
-            ],
-          },
-        },
-        {
-          path: '[*].cloudwatchLog',
-          jq: '[.[].metricFilters[] + .[].cloudwatch[] | select(.metricTransformations[].metricName  == .metric)]',
-          array_any: {
-            and: [
-              {
-                and: [
-                  {
-                    path: '[*].filterPattern',
-                    match: /\s*\$.eventName\s*=\s*ConsoleLogin\s*/,
-                  },
-                  {
-                    path: '[*].filterPattern',
-                    match: /\s*\$.errorMessage\s*=\s*"Failed authentication"\s*/,
-                  },
-                ],
-              },
-              {
-                path: '[*].sns',
-                array_any: {
-                  path: '[*].subscriptions',
-                  array_any: {
-                    path: '[*].arn',
-                    match: /^arn:aws:.*$/,
-                  },
-                },
-              },
-            ],
-          },
-        },
-      ],
-    }
+  check: ({ resource }: any): any => {
+    return resource.cloudtrail
+      ?.filter(
+        (cloudtrail: any) =>
+          cloudtrail.cloudwatchLog?.length &&
+          cloudtrail.isMultiRegionTrail === 'Yes' &&
+          cloudtrail.status.isLogging &&
+          cloudtrail.eventSelectors.some(
+            (selector: any) =>
+              selector.readWriteType === 'All' &&
+              selector.includeManagementEvents
+          )
+      )
+      .some((cloudtrail: any) => {
+        const log = cloudtrail.cloudwatchLog[0]
```

Comment:
Is there a chance there are no cloudwatchLog attached and so log would be undefined here?